py-gwpy @0.7.0: new port

duncanmmacleod opened the pull request:
#### Description

[GWpy](//gwpy.github.io/) is a python package for studying data from ground-based gravitational-wave detectors. This PR provides a Portfile for `py-gwpy` for the first time @[0.7.0](https://pypi.python.org/pypi/gwpy/0.7.0).

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->
<!-- (delete all below for minor changes) -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.13.2 17C205
Xcode 9.2 9C40b

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

pmetzger commented:
Small nit: there's no newline at the end of the file.
Another small nit: we generally (as a matter of style) don't put comments like
`# -- basic --------------------`
into our Portfiles.

Right now, Travis is misbehaving badly, so we're not likely to get a CI build for this. Can you confirm after fixing the Portfile a bit that it is building right for you and working? Then I'll do the merge.

duncanmmacleod commented:
@pmetzger, I have tried to address your comments in cd64f66, the lint and build work fine for me on my system:

```
$ port lint py27-gwpy
--->  Verifying Portfile for py27-gwpy
--->  0 errors and 0 warnings found.
$ sudo port install py27-gwpy
--->  Computing dependencies for py27-gwpy
--->  Fetching archive for py27-gwpy
Warning: Your DNS servers incorrectly claim to know the address of nonexistent hosts. This may cause checksum mismatches for some ports. See this page for more information: <https://trac.macports.org/wiki/MisbehavingServers>
--->  Attempting to fetch py27-gwpy-0.7.0_0.darwin_17.x86_64.tbz2 from https://packages.macports.org/py27-gwpy
--->  Attempting to fetch py27-gwpy-0.7.0_0.darwin_17.x86_64.tbz2 from http://mse.uk.packages.macports.org/sites/packages.macports.org/py27-gwpy
--->  Attempting to fetch py27-gwpy-0.7.0_0.darwin_17.x86_64.tbz2 from http://lil.fr.packages.macports.org/py27-gwpy
--->  Fetching distfiles for py27-gwpy
--->  Verifying checksums for py27-gwpy
--->  Extracting py27-gwpy
--->  Configuring py27-gwpy
--->  Building py27-gwpy
--->  Staging py27-gwpy into destroot
--->  Installing py27-gwpy @0.7.0_0
--->  Activating py27-gwpy @0.7.0_0
--->  Cleaning py27-gwpy
--->  Updating database of binaries
--->  Scanning binaries for linking errors
--->  No broken files found.
```

The first CI build got far enough to confirm the lint was clean, but I don't know what else was supposed to be checked before the timeout.

pmetzger commented:
Okay, I'm going to merge. Thank you for contributing this. Please monitor for failures from our build system (which you'll get as the maintainer) since the CI system is off in space.

duncanmmacleod commented:
@pmetzger, I've already received a small collection of emails complaining of `Build Failure`, linking to failures [like this](https://build.macports.org/builders/ports-10.8_x86_64_legacy-builder/builds/49650/steps/install-dependencies/logs/stdio):

```
port py27-asn1crypto  9cc52feaf47cdd2eb5ef03cdee211d9179333399d0fc34b4a39dfe8c50b0973a previously failed in build https://build.macports.org/builders/ports-10.8_x86_64_legacy-builder/builds/46865
Dependency 'py27-asn1crypto' with variants '' has previously failed and is required.
```

This seems to link back to a failure from the build for that port (not the build for `py-gwpy`). Is this something I should be worried about?

pmetzger commented:
If it's (a) for 10.8 (i.e. a quite old OS X) and (b) not your port that's failing, I wouldn't worry too much, though you can always poke the maintainer for the port your port depends on, or (if you're feeling especially public spirited!) you can see about fixing it. If it were me, I'd be relatively comfortable ignoring it given the age and the fact that it wasn't my port that was broken.

mf2k commented on python/py-gwpy/Portfile:
```diff
@@ -0,0 +1,66 @@
+name                py-gwpy
+
+categories-append   science
+maintainers         {ligo.org:duncan.macleod @duncanmmacleod}
```
Can openmaintainer be added?

pmetzger replied:
@duncanmmacleod ?

duncanmmacleod replied:
yes, what's the significance of `openmaintainer`? (@mf2k, @pmetzger)

pmetzger replied:
It means you are allowing other people to make changes when necessary without being consulted first.

pmetzger replied:
(You add it to the end of the maintainer line.)

mf2k commented on python/py-gwpy/Portfile:
```diff
@@ -0,0 +1,66 @@
+
+# until glue is available for multiple python versions, we are stuck to
+# python 2.7
+#python.versions     27 34 35 36
```
Just delete this commented python.versions line.

mf2k commented on python/py-gwpy/Portfile:
```diff
@@ -0,0 +1,66 @@
+distname            gwpy-${version}
+
+checksums   rmd160  6068b4ad15f3b3a4b615605c3d22196fb042396f \
+            sha256  6ced45ec9d62f60e01863a5ffb481d01b0f8af40333b99c0efd6db9c0f9a80f0
```
The size parameter should be added.

mf2k commented on python/py-gwpy/Portfile:
```diff
@@ -0,0 +1,66 @@
+
+master_sites        pypi:g/gwpy \
+                    https://github.com/gwpy/gwpy/releases/v${version}/
+distname            gwpy-${version}
```
Can this line be deleted? I assume the GitHub portgroup already includes this.

mf2k commented on python/py-gwpy/Portfile:
```diff
@@ -0,0 +1,66 @@
+        depends_lib-append port:py${python.version}-enum34
+    }
+
+    depends_build-append port:py${python.version}-setuptools
```
For all of the variants, you only add a dependency. What if the user has the port(s) installed but does not have the variant selected? The variant should disable the support if it is not selected.

duncanmmacleod replied:
I'm not sure how to effectively address this comment. The code is just python where certain extra class methods will work with optional dependencies, or raise `ImportError` when those are missing.

pmetzger commented:
@duncanmmacleod would you like to make those changes and do a new PR?

mf2k commented:
You can update this one and squash the changes.

duncanmmacleod commented:
@mf2k, since this has already been merged, I am happy to post a new PR that addresses your concerns. I may post questions inline above if I have them when I've thought about things more thoroughly.
